File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length())
            return false;

        vector<int> buffer(26, 0);

        for(int i = 0; i <s.length(); i++){
            buffer[int(s[i]) - int('a')] += 1;
            buffer[int(t[i]) - int('a')] -= 1;
        }

        for(auto it = buffer.begin(); it < buffer.end(); it++){
            if (*it != 0){
                return false;
            }
        }
        return true;

        }
};

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        buffer = dict()
        for idx in range(len(nums)):
            diff = target - nums[idx]
            # print(diff, buffer)
            if diff in buffer.keys():
                return [buffer[diff], idx]
            else:
                buffer[nums[idx]]=idx
        return []